### PR TITLE
Add a better error message when deps not found

### PR DESCRIPTION
**Pull request by hwright**
Use the skylark `fail` function instead of letting Python throw a
key-not-found exception.

(There may be a prettier way of doing this: I'm happy to be shown
differently, or to bikeshed the prose.)

### CODE REVIEW

**Comment by bazel-io** [comment]
Can one of the admins verify this patch?

**Comment by mattmoor** [comment]
Please test this

File: rules_python/piptool.py [hunk 1]
```diff
@@ -138,8 +138,10 @@ def pip_install():
 all_requirements = _requirements.values()
 
 def requirement(name):
-  name = name.replace("-", "_").lower()
-  return _requirements[name]
+  name_key = name.replace("-", "_").lower()
+  if name_key not in _requirements:
+    fail("Could not find pip-provided dependency: '%s'" % name)
```
Comment (mattmoor) [inline rules_python/piptool.py]: Maybe it would be interesting to include what's available?

Comment (hwright) [reply rules_python/piptool.py]: We could, but since that list is unbounded, I'm concerned that it would be too much for this error string.

**Comment by mattmoor** [comment]
Test this please

File: rules_python/piptool.py (hunk 1 above)
Comment (mattmoor) [reply rules_python/piptool.py]: This is Python, not Node.js ;-)

SGTM

**Pull request merged**